src/meta/processors/job/BalancePlan.cpp

```diff
@@ -213,7 +219,7 @@ ErrorOr<nebula::cpp2::ErrorCode, std::vector<BalanceTask>> BalancePlan::getBalan
       task.endTimeMs_ = std::get<3>(tup);
       if (resume && task.ret_ != BalanceTaskResult::SUCCEEDED) {
         // Resume the failed task, skip the in-progress and invalid tasks
-        if (task.ret_ == BalanceTaskResult::FAILED) {
+        if (task.ret_ == BalanceTaskResult::FAILED || task.ret_ == BalanceTaskResult::INVALID) {
```

Comment:
> Better check it carefully about the invalid task. IIRC, there are some cases when a task is invalid:
> 
> 1. the src is not active
> 2. the plan is canceled, and tasks which have not been run is invalid

recover a failed/stopped job is allowed, so it will try to rerun invalid/failed task

Reply:
👌